src/Domain/Services/DraftOrders.php

```diff
@@ -0,0 +1,208 @@
+	 *
+	 * Draft order status is used before payment is attempted, during checkout, when a cart is converted to an order.
+	 *
+	 * @param array $statuses Array of statuses.
+	 * @internal
+	 * @return array
+	 */
+	public function register_draft_order_status( array $statuses ) {
+		$statuses[ self::DB_STATUS ] = _x( 'Draft', 'Order status', 'woo-gutenberg-products-block' );
+		return $statuses;
+	}
+
+	/**
+	 * Register custom order post status for orders created via the API during checkout.
+	 *
+	 * @param array $statuses Array of statuses.
+	 * @internal
+
+	 * @return array
+	 */
+	public function register_draft_order_post_status( array $statuses ) {
+		$statuses[ self::DB_STATUS ] = [
+			'label'                     => _x( 'Draft', 'Order status', 'woo-gutenberg-products-block' ),
+			'public'                    => false,
+			'exclude_from_search'       => false,
+			'show_in_admin_all_list'    => false,
+			'show_in_admin_status_list' => true,
+			/* translators: %s: number of orders */
+			'label_count'               => _n_noop( 'Drafts <span class="count">(%s)</span>', 'Drafts <span class="count">(%s)</span>', 'woo-gutenberg-products-block' ),
+		];
+		return $statuses;
+	}
+
+	/**
+	 * Append draft status to a list of statuses.
+	 *
+	 * @param array $statuses Array of statuses.
+	 * @internal
+
+	 * @return array
+	 */
+	public function append_draft_order_post_status( $statuses ) {
+		$statuses[] = self::STATUS;
+		return $statuses;
+	}
+
+	/**
+	 * Delete draft orders older than a day in batches of 20.
+	 *
+	 * Ran on a daily cron schedule.
+	 *
+	 * @internal
+	 */
+	public function delete_expired_draft_orders() {
+		$count      = 0;
+		$batch_size = 20;
+		$orders     = wc_get_orders(
+			[
+				'date_modified' => '<=' . strtotime( '-1 DAY' ),
+				'limit'         => $batch_size,
+				'status'        => self::DB_STATUS,
+				'type'          => 'shop_order',
+			]
+		);
+
+		// do we bail because the query results are unexpected?
+		try {
+			$this->assert_order_results( $orders, $batch_size );
+			if ( $orders ) {
+				foreach ( $orders as $order ) {
+					$order->delete( true );
+					$count ++;
+				}
+			}
+			if ( $batch_size === $count && function_exists( 'as_enqueue_async_action' ) ) {
+				as_enqueue_async_action( 'woocommerce_cleanup_draft_orders' );
+			}
+		} catch ( Exception $error ) {
+			// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+			error_log( $error->getMessage() );
+		}
+	}
+
+	/**
+	 * Asserts whether incoming order results are expected given the query
+	 * this service class executes.
+	 *
+	 * @param WC_Order[] $order_results The order results being asserted.
+	 * @param int        $expected_batch_size The expected batch size for the results.
+	 * @throws Exception If any assertions fail, an exception is thrown.
+	 */
+	private function assert_order_results( $order_results, $expected_batch_size ) {
+		// if not an array, then just return because it won't get handled
+		// anyways.
```

Comment:
Nit: why does this comment wrap but the ones below do not?

Reply:
Habit, I wrap comments in JavaScript all the time, but sometimes remember that with PHP there's no hard rule on wrapping (so it's just a flip/flop here).